### PR TITLE
Add 2026 spring T3-1-1 skill submission for xiaoyu133133

### PR DESCRIPTION
**Pull request by xiaoyu133133**
## Competition
- Track: T3-1-1 NineToothed .skill Innovation Challenge
- Participant / Team: xiaoyu
- GitHub ID: xiaoyu133133
- Branch: 2026-spring-xiaoyu133133-t3-1-1
- Skill path: `skills/competition/ninetoothed-operator-skill/`

## Summary
This skill turns a general AI coding agent into a reliable NineToothed operator
developer. It enforces a low-variance, correctness-first workflow: classify the
task into one of four lanes, copy an existing repo operator, extract semantics,
check layout, implement minimally, verify with pytest, benchmark, self-correct via
a failure table, and pass a compliance gate.

## Scope and limitations
- Applicable: elementwise/broadcast, reduction/block, layout-sensitive, and
  performance/diagnosis/integration operator tasks.
- Out of scope: compiler-core changes, hidden-task answers, API keys, online-only
  dependencies.

## Installation and usage
See `skills/competition/ninetoothed-operator-skill/README.md` and `SKILL.md`.
Run all self-tests: `pytest skills/competition/ninetoothed-operator-skill/examples -v -p no:cacheprovider`

`pytest` output:

```
skills/.../t1-elementwise-broadcast/test_multiply.py::test[98432-dtype0-cuda] PASSED
skills/.../t2-reduction-block/test_softmax_op.py::test[1823-781-dtype0-cuda] PASSED
skills/.../t3-layout-sensitive/test_multiply_2d.py::test_contiguous[384-512-dtype0-cuda] PASSED
skills/.../t3-layout-sensitive/test_multiply_2d.py::test_non_contiguous[384-512-dtype0-cuda] PASSED
4 passed in 20.16s
```

Environment: WSL2 Ubuntu, RTX 5060, CUDA 13.0, torch 2.13.0+cu130, triton 3.7.1,
ninetoothed 0.26.0. Repo CI (ruff check, ruff format --check,
check_contributing_style.py) all pass.

## Self-test coverage
- Elementwise/broadcast (T1): elementwise multiply — 1 passed
- Reduction/block (T2): row softmax — 1 passed
- Layout-sensitive (T3): 2-D multiply, contiguous + non-contiguous — 2 passed
- Performance/diagnosis (T4): multiply benchmark + real regression diagnosis

## Benchmark summary
- multiply, 2^20 float32: torch 0.0489 ms vs NineToothed 0.0459 ms (1.07x, at parity)
- multiply, 2^24 float32: torch 0.6026 ms vs NineToothed 0.6125 ms (within noise)
- Diagnosis: an initial 22068x slowdown was traced to per-call kernel recompilation;
  fixed by compiling once (982.6 ms -> 0.0459 ms).

## With-skill vs no-skill comparison
- Without skill: agents treat the DSL like PyTorch, assume contiguous inputs, and
  skip tests; the layout-sensitive case fails silently.
- With skill: the fixed workflow forces reading a reference operator, testing a
  non-contiguous input (T3 passes), and running the CI sequence.

## Signed disclosures
- `HONOR_CODE.md`: skills/competition/ninetoothed-operator-skill/HONOR_CODE.md
- `REFERENCE.md`: skills/competition/ninetoothed-operator-skill/REFERENCE.md

## Compliance
- [x] No API keys or credentials
- [x] No hidden answers or hard-coded evaluation task names
- [x] No deleted, weakened, or bypassed tests
- [x] No fabricated benchmark results
- [x] References and AI assistance are disclosed

### CODE REVIEW

**Comment by xiaoyu133133** [comment]
很抱歉在截止时间后提交这些内容。由于网络原因，我未能在昨天及时提交比赛内容，然后官网上由于没有pr链接也没能进行提交。下面是我的提交内容，我的官网信息如下：手机号：17755182752、电子邮箱：3332643342@qq.com、姓名：兰州大学 王甘霖。十分抱歉，希望赛方可以考虑我的补交请求。
[xiaoyu_九齿skill创新挑战_提交包.zip](https://github.com/user-attachments/files/29968651/xiaoyu_.skill._.zip)
